packages/react/src/use-query.ts

```diff
@@ -67,6 +71,39 @@ export function useQuery<
   const [, forceUpdate] = useReducer(safeIncrement, 0);
   const reactionRef = useRef<Reaction>();
   const vars = useVariables(variables);
+  const { initialData } = options;
```

Comment:
we need to update the initial value for the result `useState(initialData)` as well so that the hook returns the correct value on ssr and rehydration

Reply:
@chiragpat We do, it happens when we call `trackDeps` [here](https://github.com/data-eden/data-eden/blob/423a70515d916e6ec607420f04d083770de7880f/packages/react/src/use-query.ts#L94-L106)